src/Http/Controllers/CP/ImageResizeController.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace JustBetter\ImageOptimize\Http\Controllers\CP;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Bus\Batch;
+use Illuminate\Support\Facades\Bus;
+use JustBetter\ImageOptimize\Jobs\ResizeImagesJob;
+use Statamic\Assets\AssetCollection;
+use Statamic\Assets\AssetRepository;
+use Statamic\Facades\Asset;
+use Illuminate\Support\Facades\Queue;
+
+class ImageResizeController extends Controller
+{
+    public function index() : string
+    {
+        $allAssets = Asset::all();
+        $assets = $allAssets->whereNull('image-optimized');
+
+        return view('statamic-image-optimize::cp.image-resize.index', [
+            'title' => 'JustBetter Image Optimize',
+            'total_assets' => $allAssets->count(),
+            'unoptimized_assets' => $assets->count(),
+        ]);
+    }
+
+    public function resizeImages(): array
+    {
+        $batch = (new ResizeImagesJob())->handle();
+
+        return ['imagesOptimized' => true, 'batchId' => $batch->id];
+    }
+
+    public function resizeAllImages(): array
+    {
+        $allAssets = Asset::all()->whereNotNull('image-optimized');
```

Comment:
Does it still work if we exclude the `::all()` I'm worries this will cause us to load the entire collection first and then query it. perhaps `::query()` is needed
```suggestion
        $allAssets = Asset::whereNotNull('image-optimized');
```

Reply:
The way Statamic works, this is not possible sadly. You would need to query all the asset containers first, after that you'd have to query the assets.
So i'd rather stay with this.